src/t8_vtk/t8_vtk_parallel.cxx

```diff
@@ -0,0 +1,104 @@
+/*
+This file is part of t8code.
+t8code is a C library to manage a collection (a forest) of multiple
+connected adaptive space-trees of general element classes in parallel.
+
+Copyright (C) 2015 the developers
+
+t8code is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+t8code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with t8code; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include "t8_vtk_parallel.hxx"
+
+#if T8_WITH_VTK
+#include <vtkUnstructuredGrid.h>
+#include <vtkXMLPUnstructuredGridReader.h>
+#include <vtkAppendFilter.h>
+
+vtk_read_success_t
+t8_read_parallel (const char *filename, vtkSmartPointer < vtkDataSet > grid,
```

Comment:
This whole routine looks very generic to me. :)

Isn't there already a VTK routine which does exactly this? Pass a file name and return a parallel `vtkDataSet`?

Reply:
Yes, but this function is more complex than the other one. 
Here, we also compute which proc reads which files, Then each file is assigned to the process is opened, read and merged into a single vtkDataSet. Currently this is only done by a single process (rank 0). But in the PR #608 we extend this functionality to use multiple procs to read. 

In the function `t8_read_unstructured`(which is the function you meant, I guess) a single proc opens the file and reads it. It is not extendable to be used as a parallel reader (using VTK-functions).